x/txfees/types/expected_keepers.go

```diff
@@ -2,10 +2,57 @@ package types
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	epochstypes "github.com/osmosis-labs/osmosis/v7/x/epochs/types"
 )
 
 // SpotPriceCalculator defines the contract that must be fulfilled by a spot price calculator
 // The x/gamm keeper is expected to satisfy this interface
 type SpotPriceCalculator interface {
 	CalculateSpotPrice(ctx sdk.Context, poolId uint64, tokenInDenom, tokenOutDenom string) (sdk.Dec, error)
 }
+
+// GammKeeper defines the contract needed for AccountKeeper related APIs.
+type GammKeeper interface {
+	SwapExactAmountIn(
+		ctx sdk.Context,
+		sender sdk.AccAddress,
+		poolId uint64,
+		tokenIn sdk.Coin,
+		tokenOutDenom string,
+		tokenOutMinAmount sdk.Int,
+	) (tokenOutAmount sdk.Int, spotPriceAfter sdk.Dec, err error)
+}
```

Comment:
Can we change so that GammKeeper has both SwapExactAmountIn, CalculateSpotPricce, instead of having `SpotPriceCalculator`?

Reply:
I think we should have SpotPriceCalculator be separate, since thats what other non-osmosis repositories would use. (Different handlers depend on different expected_keepers)